### ISSUE TITLE
Oracle DUAL table incorrectly quoted by RenderQuotedNames.ALWAYS

### ISSUE THREAD

**Issue by d53dave** [jOOQ/jOOQ#11790]
### Expected behavior

Executing `create.selectZero()` with the setting

```java
Settings().withRenderQuotedNames(RenderQuotedNames.ALWAYS)
```
should execute without errors.

### Actual behavior

`java.sql.SQLSyntaxErrorException: ORA-00942: table or view does not exist` is thrown. This happens because the generated SQL is:

```sql
select 0 "zero" from "dual"
```

which doesn't work. Quoting the DUAL table seems to be disallowed. Tested SQLPlus (and other clients) with Oracle 12.2.0.1.0

### Steps to reproduce the problem

See above. No code-generation required.

### Versions

- jOOQ: 3.14.8
- Java: 15.0.2
- Database (include vendor): Oracle 12.2.0.1.0
- OS: Application on MacOS, Oracle on Oracle Linux 7
- JDBC Driver (include name if inofficial driver):  ojdbc11-production

**Comment by d53dave** [comment]
Just occurred to me: the quoted name is treated as case-sensitive. 

```sql
select 0 "zero" from "DUAL"
```

works in SQLPlus.

**Comment by lukaseder** [comment]
Thank you very much for the feedback. We'll consider this in the context of https://github.com/jOOQ/jOOQ/issues/9931. Indeed, the error is not with the quoting, but with the case being lower case.

**Comment by lukaseder** [comment]
We don't have to wait for #9931 (which would also be hard to backport). Let's fix this on a per-dialect basis:

- Db2 supports `"SYSIBM"."DUAL"`, but we're already getting it right
- H2 and Oracle support `"DUAL"` in upper case
- MariaDB and MySQL only support `DUAL` (it seems to be a keyword)
- Other dialects aren't affected.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.15.0 and 3.14.10 (#11795). Thanks again for your report!
